component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPAuthenticatorCredentials.java

```diff
@@ -345,4 +351,64 @@ private boolean authorize(String secretKey, int verificationCode, long time) {
 		// Checking the validation code using the current UNIX time.
 		return checkCode(secretKey, verificationCode, time, this.config.getWindowSize());
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Check whether the verification code is valid..
+	 *
+	 * @param verificationCode Verification code.
+	 * @param username         Username.
+	 * @return whether the verification code is valid or not.
+	 */
+	public boolean isValidVerificationCode(int verificationCode, String username) {
```

Comment:
shall we rename this method to imply it store the secret key as well

Reply:
Fixed